Library/Homebrew/cask/artifact/abstract_uninstall.rb

```diff
@@ -296,6 +296,7 @@ def uninstall_script(directives, directive_name: :script, force: false, command:
         ohai "Running uninstall script #{executable}"
         raise CaskInvalidError.new(cask, "#{stanza} :#{directive_name} without :executable.") if executable.nil?
 
+        executable = File.expand_path(executable) if executable.to_s.start_with?("~")
```

Comment:
I think this logic could live inside `staged_path_join_executable`.

Reply:
This also avoids having the full user path in the error message below.

Reply:
Okay. Added changes to `abstract_artifact.rb`.